scripts/compile.py

```diff
@@ -0,0 +1,164 @@
+# Original license:
+# Copyright © 2021 rusty-snake
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# this code has been modified by https://github.com/iam-py-test, as to improve compatibility and fix issues
+
+import json
+import sys
```

Comment:
```suggestion
import json
import sys

import requests
```

Reply:
Why the big space?
I do agree that it should go at the top

Reply:
> Why the big space?

[PEP 8](https://www.python.org/dev/peps/pep-0008/) says

> Imports are always put at the top of the file, just after any module comments and docstrings, and before module globals and constants.
>
> Imports should be grouped in the following order:
>
> 1. *Standard library imports.*
> 2. *Related third party imports.*
> 3. Local application/library specific imports.
>
> *You should put a blank line between each group of imports.*

Reply:
FWIW: If you want to automate such styling things you can use [isort](https://pypi.org/project/isort/) for imports and [black](https://pypi.org/project/black/) for everything else. Or laxer formatter like [yapf](https://pypi.org/project/yapf/) or [autopep8](https://pypi.org/project/autopep8/). Sadly, they are not magic tools that turn any code into good code.

Reply:
I partially taught myself Python, so I guess I never read the standards 
Thanks